src/pages/GuestInfo.tsx

```diff
@@ -0,0 +1,159 @@
+import { Box, FormField, TextInput, DateInput, Form, Button } from 'grommet';
+import { useNavigate } from 'react-router-dom';
+import { PageWrapper } from './PageWrapper';
+import Logger from '../utils/logger';
+import { useCallback, useState } from 'react';
+import axios from 'axios';
+import { MessageBox } from 'src/components/MessageBox';
+import { DateTime } from 'luxon';
+import { useAppDispatch, useAppState } from '../store';
+import { PersonalInfo } from 'src/store/types';
+import { backend } from '../config';
+
+// const offerId = 1;
+const logger = Logger('GuestInfo');
+const defaultValue = {
+  firstname: '',
+  lastname: '',
+  birthday: new Date(),
+  email: '',
+  phone: ''
+};
+
+export const GuestInfo = () => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const { checkout } = useAppState();
+
+  const [value, setValue] = useState<PersonalInfo>(defaultValue);
+  const [loading, setLoading] = useState<boolean>(false);
+  const [error, setError] = useState<undefined | string>();
+
+  const handleSubmit = useCallback(async () => {
+    try {
+      setError(undefined);
+      setLoading(true);
+      if (
+        value.firstname === '' ||
+        value.lastname === '' ||
+        value.email === '' ||
+        value.phone === ''
+      ) {
+        throw Error('Empty fields');
+      }
+      if (checkout === undefined) {
+        throw Error('Something went wrong');
+      }
+
+      await axios.post(
+        backend.url + '/offers/' + checkout.pricedOffer.offerId + '/pii',
+        value
+      );
+      dispatch({
+        type: 'SET_CHECKOUT',
+        payload: {
+          ...checkout,
+          personalInfo: value
+        }
+      });
+
+      logger.info('Guest info sent successfully');
+      navigate('/checkout/' + checkout.pricedOffer.offerId);
+    } catch (error) {
+      const message = (error as Error).message || 'Unknown useAuthRequest error';
+      setLoading(false);
+      setError(message);
+    }
+  }, [value]);
+
+  return (
+    <PageWrapper
```

Comment:
@bobetbat PR from @silksil with refactored PageWrapper is already merged, you can update your working branch. `PageWrapper` is now renamed

Reply:
👌